File: catalog/backends/python-flask/patterns/memory_leak_service.py
from flask import request, jsonify

# Pattern: Profiling — gradual memory allocation in a cache
# Adapt: use a domain-relevant cache concept
_cache = {}


def add_to_cache():
    key = request.args.get("key", "default")
    _cache[key] = bytearray(1024 * 100)  # 100KB per entry
    return jsonify({"cached": len(_cache), "key": key})

File: catalog/backends/php-laravel/patterns/MemoryLeakService.php
<?php

use Illuminate\Http\Request;
use Illuminate\Http\JsonResponse;

// Pattern: Profiling — gradual memory allocation in a cache
// Adapt: use a domain-relevant cache concept

$cache = [];

function addToCache(Request $request): JsonResponse
{
    global $cache;
    $key = $request->query('key', 'default');
    $cache[$key] = str_repeat('x', 1024 * 100); // 100KB per entry
    return response()->json(['cached' => count($cache), 'key' => $key]);
}
